robot_smach_states/src/robot_smach_states/navigation/follow_operator2_0.py

```diff
@@ -0,0 +1,475 @@
+        smach.StateMachine.__init__(self, outcomes=['Done', 'Failed', 'Aborted'])
+        self.robot = robot
+        self.speak = speak
+        # Create the sub SMACH state machine, I think this is supposed to have different outcomes?
+        self.userdata.operator = None
+
+        with self:
+            smach.StateMachine.add('LEARN_OPERATOR', LearnOperator(robot),
+                                   transitions={'follow': 'CON_FOLLOW',
+                                                'Failed': 'Failed'},
+                                   remapping={'operator_learn_in': 'operator', 'operator_learn_out': 'operator'})
+
+            smach.StateMachine.add('ASK_FINALIZE', AskFinalize(robot),
+                                   transitions={'follow': 'CON_FOLLOW',
+                                                'Done': 'Done'})
+
+            smach.StateMachine.add('RECOVERY', Recovery(robot),
+                                   transitions={'Failed': 'Failed',
+                                                'follow': 'CON_FOLLOW'})
+
+            sm_con = smach.Concurrence(outcomes=['recover_operator', 'ask_finalize', 'keep_following'],
+                                       default_outcome='keep_following',
+                                       outcome_map={'ask_finalize': {'FOLLOWBREAD': 'no_follow_bread_ask_finalize',
+                                                                     'TRACK': 'track'},
+                                                    'recover_operator': {'FOLLOWBREAD': 'no_follow_bread_recovery',
+                                                                         'TRACK': 'no_track'}},
+                                       # 'recover_operator': {'FOLLOWBREAD': 'follow_bread',
+                                       #                     'TRACK': 'no_track'}
+                                       input_keys=['operator'])
+
+            sm_con.userdata.buffer = collections.deque()
+            sm_con.userdata.operator = None
+
+            with sm_con:
+                smach.Concurrence.add('TRACK', Track(robot), remapping={'buffer_track_in': 'buffer',
+                                                                        'buffer_track_out': 'buffer',
+                                                                        'operator_track_in': 'operator'})
+
+                smach.Concurrence.add('FOLLOWBREAD', FollowBread(robot), remapping={'buffer_follow_in': 'buffer',
+                                                                                    'buffer_follow_out': 'buffer'})
+
+            smach.StateMachine.add('CON_FOLLOW', sm_con,
+                                   transitions={'recover_operator': 'RECOVERY',
+                                                'ask_finalize': 'ASK_FINALIZE',
+                                                'keep_following': 'CON_FOLLOW'})
+
+
+# def setup_statemachine(robot):
+#     sm = smach.StateMachine(outcomes=['Done', 'Aborted'])
```

Comment:
Remove deprecated stuff (and move test, i.e., `if __name__ == "__main__"` to a different file (not in `src`)